src/XrmDefinitelyTyped/Resources/Extensions/xrm_ext_9-.d.ts

```diff
@@ -411,22 +411,19 @@ declare namespace Xrm {
         savedEntityReference: Lookup[];
     }
 
-    type PageType = "entitylist" | "webresource";
+    type PageType = "entitylist" | "entityrecord" | "webresource";
```

Comment:
Love the idea you had of extending a common interface, however it does pose a problem. Right now there is no connection between the chosen PageType and the type of extension. This means you are allowed to pass false parameters as this
```typescript
Xrm.Navigation.navigateTo({
      pageType: "entitylist",
      entityName: "",
      createFromEntity: {
        entityType: "",
        id: "",
        name: ""
      }
    })
```
The `pageType` variable should be moved to each declaration with a single allowed string that matches that declaration. For example `interface EntityList` should look like this 
```typescript
interface EntityList {
    pageType: "entitylist"
    ...
}
```
The reason why the base interface is removed is that pageType was the only common type among these interfaces. You should still keep the navigateTo function the same, with the union of the three interfaces.